### DISCUSSION TITLE
Integrating repository docs into main documentation site

### DISCUSSION THREAD

**Discussion by joewxboy** [open-horizon/open-horizon.github.io#188] [Q&A]
This discussion is meant to capture issues, questions, and requirements related to this effort.

From a high level, the tasks appear to include:

1. Identifying all repositories in this project with documentation (repo docs)
2. Creating folders in `open-horizon.github.io/docs` corresponding to the literal names of repo docs (see illustration below)
3. Ensuring repo docs are in proper format, with front matter, and in `/docs` folder (make sure the "s" is there)
4. Ensure all folders have `index.md` file and no orphans exist (orphan checking may have to happen after integration)
5. Creating GHA in repo docs to copy files over on merge containing files in /docs
6. Possibly also copy over on commits for specific branches in those branches, but this is nice-to-have
7. Possible linting action when files are copied from repo docs, nice-to-have

The initial effort was started by this issue: https://github.com/open-horizon/open-horizon.github.io/issues/29

**Comment by Sukriti-sood** [comment]
@joewxboy I have opened [issue](https://github.com/open-horizon/anax/issues/3044) as directed by you in [last comment](https://github.com/open-horizon/open-horizon.github.io/issues/29#issuecomment-1052250361) of issue #29 .

**Comment by joewxboy** [comment]
This image illustrates the `open-horizon.github.io` repo's "docs" folder tree and where the `anax` repo's "docs" folder would go in the documentation tree:
<img width="382" alt="anax-docs" src="https://user-images.githubusercontent.com/782091/156439377-d8c4baa3-0a1c-4057-9392-6e2131b2838f.png">

**Comment by joewxboy** [comment]
Status update:

1. Identifying all repositories in this project with documentation (repo docs)
    We still need to do this.  Currently we've identified `devops` and `exchange-api`.  Others?
2. **Completed.** Creating folders in open-horizon.github.io/docs corresponding to the literal names of repo docs (see illustration below)
    _This is handled automatically by the GHA._  
3. Ensuring repo docs are in proper format, with front matter, and in /docs folder (make sure the "s" is there)
    At least one repo, `exchange-api`, needs this.  Any others?
4. Ensure all folders have index.md file and no orphans exist (orphan checking may have to happen after integration)
5. **Completed.** Creating GHA in repo docs to copy files over on merge containing files in /docs
    PR has been merged.  Acceptance testing done.
6. Possibly also copy over on commits for specific branches in those branches, but this is nice-to-have
7. Possible linting action when files are copied from repo docs, nice-to-have
    Covered in related issue for developer preview GHA.  (add link)

**Reply by Sukriti-sood** [reply]
2nd one is not needed... will be done by gha.

yes, gha in the anax docs is made keeing this point.

Can you elaborate point 6.

**Reply by joewxboy** [reply]
Point 6 is confusing.  Not 100% sure what I meant. :smh: 

If I had to guess, I'd say this is about the developer preview GHA.
